java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -493,6 +576,19 @@ public void splitAndTransfer(int startIndex, int length) {
       Preconditions.checkArgument(startIndex >= 0 && length >= 0 && startIndex + length <= valueCount,
           "Invalid parameters startIndex: %s, length: %s for valueCount: %s", startIndex, length, valueCount);
       to.clear();
+      // transfer validity buffer
+      while (to.getValidityBufferValueCapacity() < length) {
+        to.reallocValidityBuffer();
+      }
+      for (int i = 0; i < length; i++) {
+        int validity = BitVectorHelper.get(validityBuffer, startIndex + i);
+        if (validity == 0) {
+          BitVectorHelper.unsetBit(to.validityBuffer, i);
+        } else {
+          BitVectorHelper.setBit(to.validityBuffer, i);
+        }
```

Comment:
Maybe we can use 
```
setValidityBit(ArrowBuf validityBuffer, int index, int value)
```
here to make the code more concise?

Reply:
done